add travis shellcheck configuration examples

jhoblitt opened the pull request:
Due to that fact that many EUPS products for 3rd party code include at least a `ups/eupspkg.cfg.sh` shell script, it is reasonable to enable `shellcheck` linting by default.

timj commented:
I do see that the ci_ packages also have build override scripts for the SCons files but it's still rare.

jhoblitt commented on stack/examples/flake8-travis.yml:
```diff
@@ -6,3 +6,19 @@ matrix:
+        docker run -v $(pwd):$(pwd) -w $(pwd) \
+          koalaman/shellcheck-alpine:v0.4.6 -x "${CHECK[@]}"
+branches:
+  only:
```
It restricts building of branches to `master`.  All PRs are still built but it avoids the slow down / waste of resources of essentially building all PR branches twice (once for the push and against for the PR).

timj commented:
I'm happy for this if the documentation that includes the example notes that the two different scripts are included and that you are allowed to pick and choose based on your particular package. @ktlim needs to give final approval.

jonathansick commented on stack/examples/flake8-travis.yml:
```diff
@@ -6,3 +6,19 @@ matrix:
+branches:
+  only:
+  - master
+notifications:
```
I don't like seeing this decision (turning off email notifications) backdoored into a single commit with an unrelated change.

I'm happy with this change personally, but this needs to be in a PR that's not titled "add shellcheck to .travis.yml example."

jhoblitt replied:
I've split the branch restriction and disabling email into #215.

jhoblitt commented:
@jonathansick Subjectively, its been a long time since I've noticed a difference between gce and docker builds.  If anything, I think they get backlogged less often.

jhoblitt commented:
Per @ktlim 's request on the jira issue, I've changed the `shellcheck` example to a stand alone configuration and also added a combined `flake8` + `shellcheck` example.

ktlim commented:
As stated in the JIRA ticket, I approve this, although also as stated in the comment in #215 , I think it is desirable in the long term for SQuaRE to take over maintaining these aspects of our GitHub code repos (much like Jonathan is responsible for enabling and maintaining LtD integration for doc repos).  That would hopefully negate the need for these examples in the dev guide.

jhoblitt commented:
Per discussion, this PR is being retracted.
